port_drayage_plugin/src/port_drayage_worker.cpp

```diff
@@ -67,6 +68,38 @@ namespace port_drayage_plugin
         _publish_mobility_operation(msg);
     }
 
+    void PortDrayageWorker::on_received_new_destination() {       
+        //  Populate the service request with the destination coordinates from the last received port drayage mobility operation message
+        cav_srvs::SetActiveRoute route_req = compose_set_active_route_request();
+
+        // Call service client to set the new active route
+        bool is_route_generation_successful = _call_set_active_route_client(route_req);
+
+        // Throw exception if route generation was not successful
+        if (!is_route_generation_successful) {
+            throw std::invalid_argument("Route generation failed. Routing could not be completed.");
+        }
+    }
+
+    cav_srvs::SetActiveRoute PortDrayageWorker::compose_set_active_route_request() const {
```

Comment:
I think generally I'd rather see this structured as taking in the data it needs as input, makes it easier to trace the data flows at run time, and makes it easier to compose unit tests for.

Reply:
Fixed